ash/src/extensions/khr/video_queue.rs

```diff
@@ -0,0 +1,220 @@
+        &self,
+        create_info: &vk::VideoSessionCreateInfoKHR<'_>,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) -> VkResult<vk::VideoSessionKHR> {
+        let mut video_session = mem::MaybeUninit::uninit();
+        (self.fp.create_video_session_khr)(
+            self.handle,
+            create_info,
+            allocation_callbacks.as_raw_ptr(),
+            video_session.as_mut_ptr(),
+        )
+        .assume_init_on_success(video_session)
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkDestroyVideoSessionKHR.html>
+    #[inline]
+    pub unsafe fn destroy_video_session(
+        &self,
+        video_session: vk::VideoSessionKHR,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_video_session_khr)(
+            self.handle,
+            video_session,
+            allocation_callbacks.as_raw_ptr(),
+        )
+    }
+
+    // Retrieve the number of elements to pass to [`get_video_session_memory_requirements`][Self::get_video_session_memory_requirements]
+    #[inline]
+    pub unsafe fn get_video_session_memory_requirements_len(
+        &self,
+        video_session: vk::VideoSessionKHR,
+    ) -> usize {
+        let mut memory_requirements_count = mem::MaybeUninit::uninit();
+        let _ = (self.fp.get_video_session_memory_requirements_khr)(
+            self.handle,
+            video_session,
+            memory_requirements_count.as_mut_ptr(),
+            ptr::null_mut(),
+        );
+        memory_requirements_count.assume_init() as _
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetVideoSessionMemoryRequirementsKHR.html>
+    ///
+    /// Call [`get_video_session_memory_requirements_len()`][Self::get_video_session_memory_requirements_len()] to query the number of elements to pass to `out`.
+    /// Be sure to [`Default::default()`]-initialize these elements and optionally set their `p_next` pointer.
+    #[inline]
+    pub unsafe fn get_video_session_memory_requirements(
+        &self,
+        video_session: vk::VideoSessionKHR,
+        out: &mut [vk::VideoSessionMemoryRequirementsKHR<'_>],
+    ) -> VkResult<()> {
+        let mut count = out.len() as u32;
+        (self.fp.get_video_session_memory_requirements_khr)(
+            self.handle,
+            video_session,
+            &mut count,
+            out.as_mut_ptr(),
+        ).result()
+    }
+
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkBindVideoSessionMemoryKHR.html>
+    #[inline]
+    pub unsafe fn bind_video_session_memory(
+        &self,
+        video_session: vk::VideoSessionKHR,
+        bind_session_memory_infos: &[vk::BindVideoSessionMemoryInfoKHR<'_>],
+    ) -> VkResult<()> {
+        (self.fp.bind_video_session_memory_khr)(
+            self.handle,
+            video_session,
+            bind_session_memory_infos.len() as _,
```

Comment:
Only `khr/acceleration_structures.rs` is erroneously using typeless casts. Please specify `u32` here explicitly.